Feature/document llama 7b on groqrack (#838)
argonne-lcf/user-guides: docs/ai-testbed/groq/examples.md

# Examples

## Running Llama-2 7B on the Groqrack

Llama-2 7B requires all 9 nodes in a Groqrack.

### Checking status

First, verify that there are no other PBS jobs using the Groqrack. This can be done from either a Groqrack node or a login node.
```console
qstat -wa
```

> If `qstat -wa` shows any jobs active, do not attempt to start Llama2.

Reviewer: the user should just wait until the command returns empty stdout?

Author: Yes.

Also check to see if anyone is connected to any of the nodes: This can be done from a either grokrack node or a login node. Logins on nodes just indicate a potential conflict; not an actual conflict. 
```console
for host in groq-r01-gn-0{1..9}; do echo $host; ssh $host /usr/bin/who; done
```

Also verify that all nodes have eight unlocked cards:

> ```

Reviewer: ```suggestion
```bash
```

for host in groq-r01-gn-0{1..9}; do ssh $host tsp-ctl status | grep -a "Device Locked" | sed "s/ Device Locked/$host Device locked/" | uniq -c ; done
```

Also, check tsp status for all cards on all nodes; verify that all cards are "Up":
```
for host in groq-r01-gn-0{1..9}; do ssh $host tsp-ctl health-check | grep -v "Health-check" | jq .[].Summary.tsp_status | sed "s/^/$host /" | uniq -c; done
```

### Running Llama2 7B

Connect to `groq-r01-gn-01.ai.alcf.anl.gov` with `ssh`:
```console
ssh  groq-r01-gn-01.ai.alcf.anl.gov
```

Copy some example files (about 25k), so that you have write access to them.
```console
mkdir ~/groq_llama2-7b-kludge/
cp /software/groq/examples/llama2-7b-kludge/*.sh ~/groq_llama2-7b-kludge/
cp /software/groq/examples/llama2-7b-kludge/*.pbs ~/groq_llama2-7b-kludge/
cd ~/groq_llama2-7b-kludge/
```

> Reserve the cluster. This will launch a placeholder PBS job that reserves the entire cluster.  It just runs `sleep 24h` on a node.   The example reserves the cluster via PBS for 2hrs.  Adjust  values as needed.

Reviewer: ```suggestion
Reserve the cluster. This will launch a placeholder PBS job that reserves the entire cluster.  It simply runs `sleep 24h` on a node. The following example script reserves the cluster via PBS for 2hrs. Adjust the values as needed.
```

# Examples

## Running Llama-2 7B on the Groqrack

> Llama-2 7B requires all 9 nodes in a groqrack.

Reviewer: ```suggestion
Llama-2 7B requires all 9 nodes in a Groqrack.
```

### Checking status

First, verify that there are no other PBS jobs using the groqrack. This can be done from either a groqrack node or a login node.
```console
qstat -wa
```
If `qstat -wa` shows any jobs active, do not attempt to start Llama2.

> Also check to see if anyone is connected to any of the nodes: This can be done from a either grokrack node or a login node. Logins on nodes just indicate a potential conflict; not an actual conflict. 

Reviewer: ```suggestion
Also, be sure to check if anyone is connected to any of the nodes. This can be done from a either Grokrack node or a login node. Logins on nodes just indicate a potential conflict; not an actual conflict. 
```

Reviewer: the last sentence is a bit confusing to me--- can you add more details?
